src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -800,26 +801,96 @@ impl ThinPool {
             None
         };
 
-        for (name, uuid, fs) in self.filesystems.iter_mut() {
-            let should_warn_on_zero = remaining_space
-                .as_ref()
-                .map(|s| *s > Sectors(0))
-                .unwrap_or(false);
-            let (needs_save, prop_diff) = fs.check(remaining_space.as_mut())?;
-            if should_warn_on_zero && remaining_space == Some(Sectors(0)) {
-                warn!(
-                    "Overprovisioning protection must be disabled to extend the filesystem further"
-                );
-            }
+        let should_warn_on_zero = remaining_space
```

Comment:
Can you just not enter the scope if ```remaining_space``` is not None and = 0? Then, the condition at the bottom of the scope is just ```!self.enable_overprov() && remaining_space = 0```?